modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java

```diff
@@ -607,6 +607,58 @@ public void testAddObjectEntry() throws Exception {
 			"{\"le\": \"2020-01-02\", \"ge\": \"2020-01-02\"}");
 
 		_assertCountAggregationObjectFieldValue(1, parentObjectEntry1);
+
+		ObjectDefinition objectDefinitionPickList = _createObjectDefinition(
+			Arrays.asList(
+				new PicklistObjectFieldBuilder(
+				).indexed(
+					true
+				).labelMap(
+					LocalizedMapUtil.getLocalizedMap(
+						RandomTestUtil.randomString())
+				).listTypeDefinitionId(
+					_listTypeDefinition.getListTypeDefinitionId()
+				).name(
+					"picklistObjectFieldName"
+				).objectFieldSettings(
+					Collections.emptyList()
+				).build()));
+
+		ListTypeEntry listTypeEntry =
+			_listTypeEntryLocalService.addListTypeEntry(
+				null, _adminUser.getUserId(),
+				_listTypeDefinition.getListTypeDefinitionId(),
+				RandomTestUtil.randomString(),
+				Collections.singletonMap(
+					LocaleUtil.US, RandomTestUtil.randomString()));
+
+		ListEntry pickListListEntry = new ListEntry();
+
+		pickListListEntry.setKey(listTypeEntry.getKey());
+		pickListListEntry.setName(listTypeEntry.getName(LocaleUtil.US));
+
+		ObjectEntry childObjectEntryListEntry = new ObjectEntry() {
+			{
+				properties = HashMapBuilder.<String, Object>put(
+					"picklistObjectFieldName", pickListListEntry
+				).build();
+			}
+		};
+
+		ObjectEntry actualObjectEntryListEntry =
+			_objectEntryManager.addObjectEntry(
+				_dtoConverterContext, objectDefinitionPickList,
+				childObjectEntryListEntry,
+				ObjectDefinitionConstants.SCOPE_COMPANY);
+
+		_assertEquals(
+			actualObjectEntryListEntry,
+			_objectEntryManager.getObjectEntry(
+				_simpleDTOConverterContext, objectDefinitionPickList,
+				actualObjectEntryListEntry.getId()));
```

Comment:
@henriquelimadotcom I'm not sure how this test is related to the issue described in [LPS-177224](https://issues.liferay.com/browse/LPS-177224).

Shouldn't you use the `PATCH` method to reproduce what is described in the ticket?

![image](https://user-images.githubusercontent.com/34186632/228032115-03210fd8-efbb-4bd9-9c13-f0d84c92bbd3.png)

Reply:
This test confirms that there is now a validation for when the picklist entry object comes with the "ListEntry" type. Before, it only checked if the type was map, if it wasn't, it would fall into else. This generated the patch bug. Having confirmation that it can arrive with the "ListEntry" type, it already avoids the patch bug and other possible bugs